core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -130,15 +130,17 @@ private[spark] class TaskSchedulerImpl private[scheduler](
 
   val mapOutputTracker = SparkEnv.get.mapOutputTracker
 
-  var schedulableBuilder: SchedulableBuilder = null
+  private val SCHEDULER_MODE_PROPERTY = "spark.scheduler.mode"
+  private var schedulableBuilder: SchedulableBuilder = null
   var rootPool: Pool = null
   // default scheduler is FIFO
-  private val schedulingModeConf = conf.get("spark.scheduler.mode", "FIFO")
+  private val schedulingModeConf = conf.get(SCHEDULER_MODE_PROPERTY, SchedulingMode.FIFO.toString)
   val schedulingMode: SchedulingMode = try {
     SchedulingMode.withName(schedulingModeConf.toUpperCase)
   } catch {
     case e: java.util.NoSuchElementException =>
-      throw new SparkException(s"Unrecognized spark.scheduler.mode: $schedulingModeConf")
+      throw new SparkException(s"Unrecognized $SCHEDULER_MODE_PROPERTY: $schedulingModeConf. " +
+        s"Supported modes: ${SchedulingMode.FAIR} or ${SchedulingMode.FIFO}.")
```

Comment:
NONE is only used in tests, and doesn't really need to be used there -- see the commit in my repo that I pointed you to earlier. Technically, removing NONE is a change in the public interface, but the docs say that only FAIR and FIFO are allowed (http://spark.apache.org/docs/latest/job-scheduling.html#configuring-pool-properties), and trying to use NONE will lead you to the "unsupported" failure, so we probably can get rid of it.